src/coverlet.core/Symbols/CecilSymbolHelper.cs

```diff
@@ -0,0 +1,290 @@
+//
+// This class is based heavily on the work of the OpenCover
+// team in OpenCover.Framework.Symbols.CecilSymbolManager
+//
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+using Coverlet.Core.Extensions;
+
+using Mono.Cecil;
+using Mono.Cecil.Cil;
+using Mono.Collections.Generic;
+
+namespace Coverlet.Core.Symbols
+{
+    public static class CecilSymbolHelper
+    {
+        private const int StepOverLineCode = 0xFEEFEE;
+        private static readonly Regex IsMovenext = new Regex(@"\<[^\s>]+\>\w__\w(\w)?::MoveNext\(\)$", RegexOptions.Compiled | RegexOptions.ExplicitCapture);
+        public static List<BranchPoint> GetBranchPoints(MethodDefinition methodDefinition)
+        {
+            var list = new List<BranchPoint>();
+            GetBranchPoints(methodDefinition, list);
+            return list;
+        }
+        private static void GetBranchPoints(MethodDefinition methodDefinition, List<BranchPoint> list)
```

Comment:
Why not just make this method signature `public static List<BranchPoint> GetBranchPoints(MethodDefinition methodDefinition)` and initialize the `BranchPoint` list in it

Reply:
It reduces the number of times we have to return the list if we need to short circuit the method.  Other than that, it's the structure that existed in OpenCover: https://github.com/OpenCover/opencover/blob/master/main/OpenCover.Framework/Symbols/CecilSymbolManager.cs#L351

Reply:
I'll appreciate if you change it, thanks